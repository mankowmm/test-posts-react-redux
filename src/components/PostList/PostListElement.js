import React, { Component } from 'react';
import { PostListButton } from './PostListButton';
import { Card, CardBody, CardTitle, CardSubtitle } from 'reactstrap';
import './postListElement.css';

export class PostListElement extends Component {
    render() {
        return (
            <div className="PostListElement">
                <Card>
                    <CardBody>
                        <CardTitle>
                            <span><b>UserId</b>: </span>
                            <span className="userIdTxt">{this.props.post.userId}</span>
                        </CardTitle>
                        <br/>
                        <CardSubtitle>
                            <span><b>Title</b>: </span>
                            <span className="titleTxt">{this.props.post.title}</span>
                        </CardSubtitle>
                        <br/>
                        <PostListButton post={this.props.post}/>
                    </CardBody>
                </Card>
            </div>
        );
    }
}

