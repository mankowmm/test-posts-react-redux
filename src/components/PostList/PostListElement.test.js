import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { shallow } from 'enzyme';
import { PostListElement } from './PostListElement';
import { MemoryRouter } from 'react-router-dom';

describe('PostListElement component test suite', ()=> {

    let elWrapper;
    let postMock;

    beforeEach(() => {
        postMock = {userId: 1, id: 1, title: 'POSTMOCK', body: 'POSTBODY'};
    })
    
    it('should validate with snapshot', () => {
        const tree = renderer
          .create(<MemoryRouter><PostListElement post={postMock}/></MemoryRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });      
    
    it('should contain PostListButton', ()=> {
        elWrapper = mount(<MemoryRouter><PostListElement post={postMock}/></MemoryRouter>).find(PostListElement);
        expect(elWrapper.find('PostListButton').length).toBe(1);
    })

    it('should render elements from post object', ()=> {
        elWrapper = mount(<MemoryRouter><PostListElement post={postMock}/></MemoryRouter>).find(PostListElement);
        expect(elWrapper.find('.userIdTxt').text()).toBe(postMock.userId.toString());
        expect(elWrapper.find('.titleTxt').text()).toBe(postMock.title); 
    })

});