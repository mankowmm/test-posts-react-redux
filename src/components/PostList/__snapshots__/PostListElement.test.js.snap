// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListElement component test suite should validate with snapshot 1`] = `
<div
  className="PostListElement"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        <span>
          <b>
            UserId
          </b>
          : 
        </span>
        <span
          className="userIdTxt"
        >
          1
        </span>
      </h5>
      <br />
      <h6
        className="card-subtitle"
      >
        <span>
          <b>
            Title
          </b>
          : 
        </span>
        <span
          className="titleTxt"
        >
          POSTMOCK
        </span>
      </h6>
      <br />
      <a
        className="btn btn-secondary"
        href="/post/1"
        onClick={[Function]}
      >
        See detail
      </a>
    </div>
  </div>
</div>
`;
